modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/coercion/NumericArithmeticsExecutionTest.java

```diff
@@ -0,0 +1,808 @@
+        String sql = "SELECT c1 / c2 FROM t";
+        Assumptions.assumeFalse(typePair.first() instanceof DecimalNativeType || typePair.second() instanceof DecimalNativeType,
+                "need to be fixed after: https://issues.apache.org/jira/browse/IGNITE-23171");
+
+        try (ClusterWrapper testCluster = testCluster(typePair, binOpDataProvider(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("moduloArgs")
+    public void moduloOp(TypePair typePair, Matcher<Object> resultMatcher) throws Exception {
+        // modulo is undefined for fractional types
+        if (APPROXIMATE_NUMERIC_TYPES.contains(typePair.first()) || APPROXIMATE_NUMERIC_TYPES.contains(typePair.second())) {
+            return;
+        }
+
+        String sql = "SELECT c1 % c2 FROM t";
+
+        try (ClusterWrapper testCluster = testCluster(typePair, eqDataProvider(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @Test
+    public void checkAllTypePairs() {
+        checkIncludesAllNumericTypePairs(sumArgs());
+        checkIncludesAllNumericTypePairs(subtractArgs());
+        checkIncludesAllNumericTypePairs(multArgs());
+        checkIncludesAllNumericTypePairs(divArgs());
+        checkIncludesAllNumericTypePairs(moduloArgs());
+    }
+
+    private static DataProvider<Object[]> binOpDataProvider(TypePair typePair) {
+        Object val1 = SqlTestUtils.generateValueByType(typePair.first());
+        Object val2 = SqlTestUtils.generateValueByType(typePair.second());
+
+        return DataProvider.fromRow(new Object[]{0, val1, val2}, 1);
+    }
+
+    private static Stream<Arguments> sumArgs() {
+        return Stream.of(
+                forTypePair(NumericPair.TINYINT_TINYINT).resultWillBe(NativeTypes.INT8),
+                forTypePair(NumericPair.TINYINT_SMALLINT).resultWillBe(NativeTypes.INT16),
+                forTypePair(NumericPair.TINYINT_INT).resultWillBe(NativeTypes.INT32),
+                forTypePair(NumericPair.TINYINT_BIGINT).resultWillBe(NativeTypes.INT64),
+                forTypePair(NumericPair.TINYINT_DECIMAL_1_0).resultWillBe(Types.DECIMAL_2_0),
+                forTypePair(NumericPair.TINYINT_DECIMAL_2_1).resultWillBe(Types.DECIMAL_3_1),
+                forTypePair(NumericPair.TINYINT_DECIMAL_4_3).resultWillBe(Types.DECIMAL_5_3),
+                forTypePair(NumericPair.TINYINT_DECIMAL_2_0).resultWillBe(Types.DECIMAL_2_0),
+                forTypePair(NumericPair.TINYINT_DECIMAL_3_1).resultWillBe(Types.DECIMAL_3_1),
+                forTypePair(NumericPair.TINYINT_DECIMAL_5_3).resultWillBe(Types.DECIMAL_5_3),
+                forTypePair(NumericPair.TINYINT_DECIMAL_5_0).resultWillBe(Types.DECIMAL_2_0),
+                forTypePair(NumericPair.TINYINT_DECIMAL_6_1).resultWillBe(Types.DECIMAL_3_1),
+                forTypePair(NumericPair.TINYINT_DECIMAL_8_3).resultWillBe(Types.DECIMAL_5_3),
```

Comment:
it looks suspiciously somthing + dec(8.3) = dec(5.3). I will still suggest using generate values with real precision and scale, otherwise, we are checking not we want.

Reply:
not relevant for now